Tribler/Core/Modules/process_checker.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+from Tribler.Core.SessionConfig import SessionStartupConfig
+
+
+LOCK_FILE_NAME = 'tribler.lock'
+
+
+class ProcessChecker(object):
+    """
+    This class contains code to check whether a Tribler process is already running.
+    """
+
+    @staticmethod
+    def is_pid_running(pid):
+        """
+        Check whether a given process ID is currently running.
+        """
+        try:
+            os.kill(pid, 0)
+        except OSError:
+            return False
+        else:
+            return True
+
+    def create_lock_file(self):
+        """
+        Create the lock file and write the PID in it. We also create the directory structure since the ProcessChecker
+        might be called before the .Tribler directory has been created.
+        """
+        if not os.path.exists(self.statedir):
+            os.makedirs(self.statedir)
+
+        with open(self.lock_file_path, 'wb') as lock_file:
+            lock_file.write(str(os.getpid()))
+
+    def remove_lock_file(self):
+        """
+        Remove the lock file.
+        """
+        os.unlink(self.lock_file_path)
+
+    def get_pid_from_lock_file(self):
+        """
+        Returns the PID from the lock file.
+        """
+        with open(self.lock_file_path, 'rb') as lock_file:
+            return lock_file.read()
+
+    def __init__(self):
+        """
+        Check whether a lock file exists in the Tribler directory. If not, create the file. If it exists,
+        check the PID that is written inside the lock file.
+        """
+        self.already_running = False
+
+        self.statedir = SessionStartupConfig().get_state_dir()
+        self.lock_file_path = os.path.join(self.statedir, LOCK_FILE_NAME)
+
+        if os.path.exists(self.lock_file_path):
+            file_pid = self.get_pid_from_lock_file()
+            if file_pid == str(os.getpid()):
```

Comment:
In this case, we're checking for value equality and not object equality so I think it's safer to use `==`. See for instance the following example:

```
>>> p = ''.join(['a', 'b'])
>>> p is 'ab'
False
>>> p == 'ab'
True
```